src/PKSim.Core/Snapshots/Mappers/ExtendedPropertyMapper.cs

```diff
@@ -0,0 +1,72 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using OSPSuite.Core.Domain;
+using OSPSuite.Utility.Extensions;
+using ModelExtendedProperty = OSPSuite.Core.Domain.IExtendedProperty;
+using SnapshotExtendedProperty = PKSim.Core.Snapshots.ExtendedProperty;
+
+namespace PKSim.Core.Snapshots.Mappers
+{
+   public class ExtendedPropertyMapper : SnapshotMapperBase<ModelExtendedProperty, SnapshotExtendedProperty>
+   {
+      public override SnapshotExtendedProperty MapToSnapshot(ModelExtendedProperty extendedProperty)
+      {
+         return SnapshotFrom(extendedProperty, snapshot =>
+         {
+            snapshot.ListOfValues = extendedProperty.ListOfValuesAsObjects.Any() ? extendedProperty.ListOfValuesAsObjects.ToList() : null;
+            snapshot.Value = extendedProperty.ValueAsObject;
+            snapshot.Type = extendedProperty.Type;
+            snapshot.Description = SnapshotValueFor(extendedProperty.Description);
+            snapshot.Name = extendedProperty.Name;
+            snapshot.FullName = extendedProperty.FullName;
+            snapshot.DisplayName = extendedProperty.DisplayName;
+            snapshot.ReadOnly = extendedProperty.ReadOnly;
+         });
+      }
+
+      public override ModelExtendedProperty MapToModel(SnapshotExtendedProperty snapshot)
```

Comment:
That's where I initially tried to use generics, but was unsuccessful in creating a mapper and getting it to resolve properly.